### ISSUE TITLE
Writing is not compatible with np.ndarrays

### ISSUE THREAD

**Issue by ameliesc** [schipiga/pandas-tfrecords#12]
The current solution did not work for np.ndarrays.

For writing I used the following workaround: 

```
def _bytes_feature(value, ):
    if isinstance(value, type(tf.constant(0))):
        value = value.numpy()
    if isinstance(value, str):
        value = value.encode(encoding='UTF-8')
    if isinstance(value, np.ndarray):
        return _bytes_feature(serialize_array(value))
    return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))


def _float_feature(value):
    return tf.train.Feature(float_list=tf.train.FloatList(value=[value]))


def _int64_feature(value):
    return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))


def serialize_array(array):

    array = tf.io.serialize_tensor(array)
    return array
```

It becomes more complicated when trying to read as the datatype gets lost. My workaround has been to derive the schema  from the pandas dataframe and to write it to a json which looks something like this:

`{"url": "str", "article": "str", "label": "str", "publisher": "str", "split": "str", "bag_of_words": "int"}`

Then when reading I would pass this schema and I changed the `_get_feature_type` function to work with strings rather than the actual datatype as we can't serialise types to json.

**Comment by schipiga** [comment]
hi @ameliesc thank you for report. Could you please provide data example which you can't convert to tfrecords via this library
